src/main/java/org/wso2/carbon/esb/connector/SessionManager.java

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright (c) 2018, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.wso2.carbon.esb.connector;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.jsmpp.session.BindParameter;
+import org.jsmpp.session.SMPPSession;
+
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+
+public class SessionManager {
+    protected Log log = LogFactory.getLog(this.getClass());
+    private Map<String, SMPPSession> smppSessionList;
+    private static SessionManager sessionManager;
+
+    private SessionManager() {
+        smppSessionList = new HashMap();
+    }
+
+    public static SessionManager getInstance() {
+        if (sessionManager == null)
```

Comment:
Don't we need to use synchronization block while creating singlton

Reply:
agreed.